### ISSUE TITLE
[naming-convention] False positive when the enums is declared in uppercase

### ISSUE THREAD

**Issue by jmakelele** [typescript-eslint/typescript-eslint#3525]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/naming-convention": ["error", {
      selector: 'enum', /* or 'typeLike' */
      format: ['PascalCase']
    }]
  }
}
```

```TS
export enum ERRORS {
  CONNECTION = 'Connection',
  TECHNICAL = 'Technical'
}
```

**Expected Result**

Warnings for ERRORS

**Actual Result**

No warning

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.26.1` |
| `@typescript-eslint/parser`        | `4.26.1` |
| `TypeScript`                       | `4.3.2` |
| `ESLint`                           | `7.28.0` |
| `node`                             | `14.15.0` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md#format

Please read the rule docs.
`StrictPascalCase` will be of interest to you.

**Comment by jmakelele** [comment]
Thank you for your help.
